src/diagnostics/core_diagnostics.jl

```diff
@@ -1349,3 +1349,95 @@ add_diagnostic_variable!(
         end
     end,
 )
+
+###
+# Convective Available Potential Energy (2d)
+###
+function compute_cape!(out, state, cache, time)
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    g = TD.Parameters.grav(thermo_params)
+
+    # Get surface parcel properties
+    surface_thermal_state = Fields.level(cache.precomputed.ᶜts, 1)
+    surface_q = TD.total_specific_humidity.(thermo_params, surface_thermal_state)
+    surface_θ_liq_ice = TD.liquid_ice_pottemp.(thermo_params, surface_thermal_state)
+
+    # Create parcel thermodynamic states at each level based on energy & moisture at surface
+    parcel_ts_moist = TD.PhaseEquil_pθq.(thermo_params, 
+                                        cache.precomputed.ᶜts.p, 
```

Comment:
Don't understand here. You are passing in pressure at each level but first cell center theta and q?